deploy/base/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: objectstorage-provisioner
+  namespace: objectstorage-provisioner-ns
+  labels:
+    app.kubernetes.io/part-of: container-object-storage-interface
+    app.kubernetes.io/component: provisioner
+    app.kubernetes.io/version: main
+    app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+spec:
+  replicas: 1
+  minReadySeconds: 30
+  progressDeadlineSeconds: 600
+  revisionHistoryLimit: 3
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  selector:
+    matchLabels:
+      app.kubernetes.io/part-of: container-object-storage-interface
+      app.kubernetes.io/component: provisioner
+      app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/part-of: container-object-storage-interface
+        app.kubernetes.io/component: provisioner
+        app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+    spec:
+      serviceAccountName: objectstorage-provisioner-sa
+      containers:
+      - name: objectstorage-sample-driver
```

Comment:
Not sure this repository should have a `deploy/base` in the first place, given you can't really deploy the sidecar in isolation. This is more the deployment of the sample driver... Hence a more appropriate name could be in order.

Reply:
@wlan0 @rrati suggestions?